tests/sqllogictests/suites/base/05_ddl/05_0000_ddl_create_tables

```diff
@@ -332,6 +332,12 @@ create table t_with_bloom_index(a int, b int) bloom_index_columns='b'
 statement ok
 drop table if exists t_with_bloom_index;
 
+statement ok
+create table t_nest_type(a int, b array(int not null));
```

Comment:
test case should include three types: `array(int not null)`, `array(int null)` and `array(int)`

and please also add some unit tests for parser in this file https://github.com/datafuselabs/databend/blob/main/src/query/ast/tests/it/parser.rs